### ISSUE TITLE
fix database import for cases where the owner is not one of the expected ones

### ISSUE THREAD

**Issue by eonrider** [nextcloud/all-in-one#967]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Perform step 12 of migration instructions. https://github.com/nextcloud/all-in-one/blob/v1.5.0/migration.md
2. See `ERROR:  role "ncadmin" does not exist` everywhere during database dump import

### Expected behavior 
The import should be successful.

### Actual behavior 
The import errors out because the database role does not exist.

### Host OS 
Raspberry Pi OS Lite (64-bit)

#### Nextcloud AIO version 
Nextcloud AIO v1.5.0

#### Current channel 
`latest-arm64`

#### Other valuable info 
It seems to me that the expected role is actually `nextcloud` as seen here https://github.com/nextcloud/all-in-one/blob/v1.5.0/php/containers.json#L72. Instead, my script tries to use `ncadmin` as that's what I used when I converted my MariaDB database to Postgres using `export PG_USER="ncadmin"` as suggested by the migration document.

**Comment by szaimen** [comment]
Hi, thnaks for your report. Can you please run `grep 'Owner:'` on your database export and post the output here?

I think we rather need to adjust the things here: https://github.com/nextcloud/all-in-one/blob/e6a9fcd3c87f0ca055688b67c14c1d3651ee56b2/Containers/postgresql/start.sh#L68-L84

**Comment by szaimen** [comment]
I suppose it should return `Owner: nextcloud` but not really sure. So that is why I am asking...

**Comment by eonrider** [comment]
Yes. Here's just a snippet:
```
-- Name: oc_accounts; Type: TABLE; Schema: public; Owner: ncadmin
-- Name: oc_accounts_data; Type: TABLE; Schema: public; Owner: ncadmin
-- Name: oc_accounts_data_id_seq; Type: SEQUENCE; Schema: public; Owner: ncadmin
-- Name: oc_accounts_data_id_seq; Type: SEQUENCE OWNED BY; Schema: public; Owner: ncadmin
-- Name: oc_activity; Type: TABLE; Schema: public; Owner: ncadmin
-- Name: oc_activity_activity_id_seq; Type: SEQUENCE; Schema: public; Owner: ncadmin
-- Name: oc_activity_activity_id_seq; Type: SEQUENCE OWNED BY; Schema: public; Owner: ncadmin
-- Name: oc_activity_mq; Type: TABLE; Schema: public; Owner: ncadmin
-- Name: oc_activity_mq_mail_id_seq; Type: SEQUENCE; Schema: public; Owner: ncadmin
-- Name: oc_activity_mq_mail_id_seq; Type: SEQUENCE OWNED BY; Schema: public; Owner: ncadmin
-- Name: oc_addressbookchanges; Type: TABLE; Schema: public; Owner: ncadmin
-- Name: oc_addressbookchanges_id_seq; Type: SEQUENCE; Schema: public; Owner: ncadmin
-- Name: oc_addressbookchanges_id_seq; Type: SEQUENCE OWNED BY; Schema: public; Owner: ncadmin
-- Name: oc_addressbooks; Type: TABLE; Schema: public; Owner: ncadmin
-- Name: oc_addressbooks_id_seq; Type: SEQUENCE; Schema: public; Owner: ncadmin
-- Name: oc_addressbooks_id_seq; Type: SEQUENCE OWNED BY; Schema: public; Owner: ncadmin
-- Name: oc_appconfig; Type: TABLE; Schema: public; Owner: ncadmin
-- Name: oc_authorized_groups; Type: TABLE; Schema: public; Owner: ncadmin
...
```

**Comment by szaimen** [comment]
I see, thanks! So we need to adjust the logic there.

Here an example line we could grep for during the script:
 `-- Name: oc_appconfig; Type: TABLE; Schema: public; Owner: oc_nextcloud`

**Comment by szaimen** [comment]
As for a workaround you may solve it alreaddy by renaming `Owner: nextcloud` to `Owner: oc_nextcloud`.

E.g. with sed -i `s|Owner: nextcloud$|Owner: oc_nextcloud|' database-dump.sql`

**Comment by eonrider** [comment]
> As for a workaround you may solve it alreaddy by renaming `Owner: ncadmin` to `Owner: oc_nextcloud`.
> 
> E.g. with `sed -i 's|Owner: ncadmin$|Owner: oc_nextcloud|' database-dump.sql`



> sed -i 's|Owner: ncadmin$|Owner: oc_nextcloud|' database-dump.sql

I suppose these ones would need to be updated as well?
```
$ grep 'ncadmin' database-dump.sql
ALTER TABLE public.oc_accounts OWNER TO ncadmin;
ALTER TABLE public.oc_accounts_data OWNER TO ncadmin;
ALTER TABLE public.oc_accounts_data_id_seq OWNER TO ncadmin;
ALTER TABLE public.oc_activity OWNER TO ncadmin;
ALTER TABLE public.oc_activity_activity_id_seq OWNER TO ncadmin;
ALTER TABLE public.oc_activity_mq OWNER TO ncadmin;
ALTER TABLE public.oc_activity_mq_mail_id_seq OWNER TO ncadmin;
ALTER TABLE public.oc_addressbookchanges OWNER TO ncadmin;
ALTER TABLE public.oc_addressbookchanges_id_seq OWNER TO ncadmin;
ALTER TABLE public.oc_addressbooks OWNER TO ncadmin;
ALTER TABLE public.oc_addressbooks_id_seq OWNER TO ncadmin;
...
```

**Comment by szaimen** [comment]
> I suppose these ones would need to be updated as well?

Yes  👍

**Comment by dantefromhell** [comment]
I ran into the same issue when migrating.

It was coupled with another issue:
the `pg_dump` command in step 4) only exports the database contents but not the roles (postgresql user accounts).
As a result the import fails because the `owner` of tables does not exist.**

I used the approach describes here [1] to dump the roles into a separate file, than did a `cat roles_dump_file tables_dump_file > database-dump.sql`. The combined dump successfully loaded during import.

**I guesstimate this is only a problem if you try to restore into a postgres instance that has never been initialized before. But that was exactly my situation, I imported the old data during the first startup of `nextcloud-aio`.

[1] https://dba.stackexchange.com/questions/20986/postgresql-how-do-i-dump-and-restore-roles-for-a-cluster
